### ISSUE TITLE
Segmentation fault when using get_drawings()

### ISSUE THREAD

**Issue by Luux** [pymupdf/PyMuPDF#1971]
After upgrading from `pymupdf==1.19.6` to `pymupdf==1.20.2`, our test pipelines always resulted in segfaults. After tinkering around, I managed to isolate the bug with a minimal sample:

When using `get_drawings()` on certain documents, this reproducably results in a segfault if `get_drawings` was used with some other document before:

1. Load document A
2. get_drawings of A
3. unload A
4. load faulty document B
5. get_drawings of B
6. segfault
 
 This seems to happen with *some* documents only. I observed that both
1. Load document A
2. get_drawings of A
3. unload A
4. load document A
5. get_drawings of A

and

1. Load faulty document B
2. get_drawings of B
3. unload B
4. load document A
5. get_drawings of A

may work just fine.

The safest variant to reproduce is to just
1. Load faulty document B
2. get_drawings of B (still works)
3. unload B
4. load faulty document B
5. get_drawings of B
6. segfault

For the segfault, faulty document B needs to be loaded as *second* document. Unloading via `B.close()` is optional & does not alter the result.

## To Reproduce (mandatory)
Run
```
import fitz
import pathlib

PDF_CROP = pathlib.Path("crop.pdf")

for _ in range(2):
    doc = fitz.Document(PDF_CROP)
    page = next(doc.pages())
    page.get_drawings()
    doc.close()
```
with the crop attached:  
[crop.pdf](https://github.com/pymupdf/PyMuPDF/files/9799426/crop.pdf)

## Expected behavior (optional)
No segfault.

## Your configuration (mandatory)
 - (K)Ubuntu 22.04.1 LTS
 - Python 3.10.4
 - pymupdf==1.20.2, installed through pip

**Comment by julian-smith-artifex-com** [comment]
Thanks for this bug report and for the simple reproducer.

We have a fix, but it needs cleaning up a little and i haven't run the full test suite on it yet. Hopefully it'll be ok and this bug will be fixed in the next release.

**Comment by julian-smith-artifex-com** [comment]
Fixed in 1.21.0
